### ISSUE TITLE
after changing mac with macchanger, d4 doesn't connect to wifi anymore

### ISSUE THREAD

**Issue by norayr** [maemo-leste/bugtracker#707]
even if restarting icd2.
to reproduce:

* disconnect from current wifi access point, if connected
* go to offline mode
* macchanger -a (or -r) wlan0
* go to online mode
* sudo service icd2 restart
* try to connect to any wifi access point

**Comment by IMbackK** [comment]
pretty mutch expected, icd only uses wlan0, udev decides what device is wlan0 by saveing the mac address, if you change it, after reboot wlan0 becomes wlan1 and its over

**Comment by norayr** [comment]
well i think it worked some time ago because maemo wiki explains how to use macchanger: https://leste.maemo.org/Tricks

**Comment by MerlijnWajer** [comment]
Yeah, so this probably comes down to libicd-network-wpasupplicant wanting to work on `wlan0` only currently. We can change this, but it raises a few other questions and things I'd have to investigate, like if the code can currently re-use the same wpa_supplicant instance, or if we'd have multiple, and if we really want the network interface names to linearly increase every time you change the MAC.

We could just point the udev rules file to /dev/null and at least this specific problem would be gone. (I do think we want to support other names than wlan0, fwiw)

**Comment by norayr** [comment]
wait, i don't understand why are you saying about wlan1, wlan2 etc.

i just again changed mac address of wlan0:

```
macchanger -a wlan0
```

it wrote on console current, permanent and new macs.
```
ifconfig -a
```
does not reveal any `wlan1` or `wlan2`, just mac address of the same old `wlan0` got changed.
